pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -234,6 +231,23 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		return *result, err
 	}
 
+	var tlsSecret *corev1.Secret
+	if infinispan.IsEncryptionEnabled() {
+		tlsSecret = &corev1.Secret{}
+		if result, err := kube.LookupResource(infinispan.GetEncryptionSecretName(), infinispan.Namespace, tlsSecret, r.client, reqLogger); result != nil {
+			return *result, err
+		}
+
+		// Set the controller reference to be the Infinispan CR so that we can receive reconcile events via EnqueueRequestForOwner
+		res, err := controllerutil.CreateOrUpdate(ctx, r.client, tlsSecret, func() error {
+			return controllerutil.SetControllerReference(infinispan, tlsSecret, r.scheme)
```

Comment:
@ryanemerson have you validate that second call will not cause an exception. Seems to need additional logic like
`!metav1.IsControlledBy(..., ...)` before, but not sure

Reply:
@dmvolod I don't think this is necessary do to the `referSameObject` check at L85 in `SetControllerReference`.

Reply:
Oh, yes, sure. Hope Secret will be removed by garbage collector and this situation never happens with new Infinispan CR

Reply:
If someone will try to share Secret between clusters conflict could happens and operator will fails with the `SetControllerReference` as this is just name.